tasks/clean.yaml

```diff
@@ -0,0 +1,13 @@
+tasks:
+  - name: clean-runner
+    description: Free GH Runner space
+    actions:
+      - description: Free GH Runner space
+        cmd: |
+          df -h
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /opt/ghc
+          sudo rm -rf /opt/hostedtoolcache/CodeQL
+          sudo docker image prune --all --force
+          df -h
```

Comment:
Wonder if we should wrap this in a `if ${CI}` or one of the other predefined envs so that it doesn't bork someone's stuff locally - https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

Reply:
Converting it into an action, so the `if ${CI}` is no longer necessary